modal/runner.py

```diff
@@ -521,6 +525,9 @@ async def _deploy_app(
 
     t0 = time.time()
 
+    # Get git information to track deployment history
+    commit_info_future = get_git_commit_info()
```

Comment:
It looks like you're trying to to start this future early to let it run concurrently with the code afterward. I think you have to wrap it in `asyncio.create_task()` to make this work. Unlike JS, Python's asyncio doesn't schedule futures unless they're being awaited, or they're wrapped in an asyncio.Task